### DISCUSSION TITLE
ClientSecret or SecretHash on login

### DISCUSSION THREAD

**Discussion by apotts-afg** [aws/aws-sdk-net-extensions-cognito#91] [Q&A] [answered]
I don't see a way to include the SecretHash as part of the `InitiateSrpAuthRequest`, how do I include that with this extension?

**Comment by apotts-afg** [comment]
So, I did see how to initialize the ClientSecret by setting it on the UserPool, however in the call to `CreateSrpAuthRequest`, `this.ClientSecret` is still null

**Comment by ashishdhingra** [comment, marked as answer]
Hi @anthony-potts,

Thanks for posting guidance question. Could you please share the sample code to troubleshoot the issue?

Regards,
Ashish

[a comment by an automated account is not shown]
